src/database/database.manager.ts

```diff
@@ -0,0 +1,68 @@
+import { SupabaseClient } from '@supabase/supabase-js/src'
+
+export class DatabaseManager {
+  private supabase: SupabaseClient
+
+  constructor(supabase: SupabaseClient) {
+    this.supabase = supabase
+  }
+
+  setSupabase(supabase: SupabaseClient) {
+    this.supabase = supabase
+  }
+
+  async fetchData(tableName: string): Promise<{ [p: string]: any }[] | null> {
+    const { data, error } = await this.supabase.from(tableName).select()
+    console.error(error)
+    return data
+  }
+
+  async fetchDataWWhere(
+    tableName: string,
+    filter: SupabaseFilter
+  ): Promise<{ [p: string]: any }[] | null> {
```

Comment:
Please type correctly or suppress warning if there is no type for this in the supabase package